### PR TITLE
chore: add eslint plugin vuejs accessibility

### PR DESCRIPTION
**Pull request by lucianmurmurache**
### Contributor checklist

- [x] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch

---

### Description
Implement [eslint-plugin-vuejs-accessibility](https://www.npmjs.com/package/eslint-plugin-vuejs-accessibility).
Address issues flagged by the plugin.
Update docs to reflect the above updates.

### Related issue

- #576

### CODE REVIEW

**Comment by andrewtavis** [comment]
Nice @lucianmurmurache! Is there a way that we can also add this into the `pr_frontend_check` workflow?

**Comment by lucianmurmurache** [comment]
> Nice @lucianmurmurache! Is there a way that we can also add this into the `pr_ci_frontend` workflow?

@andrewtavis it's already part of the flow. For example this [run](https://github.com/activist-org/activist/actions/runs/8222891255) failed due to a11y checks (happy little accident 🎨🖌️) `eslint . --ext .js,.vue` runs all the listed checks in [.eslintrc.js](https://github.com/activist-org/activist/blob/bc488886456c7716eaf2a35717e2530d519fa6c1/frontend/.eslintrc.js), and I added the plugin to the [eslint module](https://github.com/activist-org/activist/pull/755/files#diff-093b2d5625f20b6d845fe73f1af12762fc64ba9d0b9d94b14e8aa287e2d2ed1dR4). Also your editor will scream errors at you for :accessibility: violations instead of warnings.

**Comment by andrewtavis** [comment]
This is amazing 🤩 Thanks @lucianmurmurache! :) :)

**Comment by lucianmurmurache** [comment]
@andrewtavis I can make a little change here and also solve #338 in the same PR if you agree to a 2in1 approach

**Comment by andrewtavis** [comment]
Either or, @lucianmurmurache! Happy to get another PR, but feel free to send it along to this one as well :) Goal is to merge by tomorrow :)

File: frontend/components/SearchBar.vue [hunk 1]
```diff
@@ -66,11 +73,13 @@
...
       size="1em"
     />
+    <label for="expanded-search-input">{{ $t("_global.search") }}</label>
     <input
       v-if="expanded"
+      id="expanded-search-input"
       class="bg-transparent focus:outline-none"
       type="text"
-      placeholder="Search"
+      :placeholder="$t('_global.search')"
```
Comment (andrewtavis) [inline frontend/components/SearchBar.vue:82]: Nice catch, thanks! 😎

File: frontend/components/card/discussion/CardDiscussionInput.vue [hunk 2]
```diff
@@ -109,7 +109,9 @@
         <div class="flex items-center space-x-3">
           <Button
             @mouseenter="showTooltip = true"
+            @focus="showTooltip = true"
```
Comment (andrewtavis) [inline frontend/components/card/discussion/CardDiscussionInput.vue:112]: Nice, yes definitely needed :)

File: frontend/components/sidebar/left/SidebarLeft.vue [hunk 3]
```diff
@@ -4,10 +4,20 @@
...
+    "
     @mouseleave="
       sidebar.collapsed = true;
       setContentScrollable();
     "
+    @blur="
+      sidebar.collapsed = true;
+      setContentScrollable();
+    "
+    role="menu"
```
Comment (andrewtavis) [inline frontend/components/sidebar/left/SidebarLeft.vue:19]: Minor note here: as we are switching from the activist logo to an icon when we switch to a collapsed sidebar we are losing focus on blur when we tab from the sidebar itself to the logo :) Do you want to send along a quick fix for this?

Note that I have a commit coming with some changes shortly 😊

File: frontend/components/menu/MenuSearchResult.vue [hunk 4]
```diff
@@ -1,7 +1,10 @@
 <template>
   <div
     @mouseleave="showTooltip = false"
+    @blur="showTooltip = false"
```
Comment (andrewtavis) [inline frontend/components/menu/MenuSearchResult.vue:4]: I'm gonna revert these changes as it's actually not the desired behavior :) This div is to add the `p-2` around the button or the tooltip if it's shown and allow for the mouse leave only, but shouldn't be focused itself. The focus should go directly to the button. If we need these in here for a11y reasons, then let's discuss it in a sync as this component is actually pretty fine tuned for usability 😊

File: frontend/components/sidebar/left/SidebarLeft.vue (hunk 3 above)
Comment (andrewtavis) [reply frontend/components/sidebar/left/SidebarLeft.vue:19]: As this is sidebar related, maybe we can add this into #338 so we can merge this in, @lucianmurmurache :) :)

**Comment by andrewtavis** [comment]
Hehe ok then, let me revert some of my changes 😅 Why aren't these showing up in ESLint though, @lucianmurmurache?

**Comment by andrewtavis** [comment]
As of now I'm adding the following to `MenuSearchResult` as I'm really not thinking that there's an easy solution for this:

```html
  <!-- eslint-disable-next-line vuejs-accessibility/mouse-events-have-key-events, vuejs-accessibility/no-static-element-interactions -->
```

I'm actually not getting the `Anchors must have content and the content must be accessible by a screen reader` errors locally though 🤔 Seems like this could be fixed in a few ways. Let me know if you have suggestions on this, @lucianmurmurache!

**Comment by andrewtavis** [comment]
054024a sends along the final fixes. We needed some aria labels for `<a>` tags. I did end up ignoring the ESLint errors for `MenuSearchResult` for now. If we want to work on that, then we should likely talk about it in a dev sync before so I can go over the functionality 😊

**Review by andrewtavis — APPROVED** [review]
This is a huge step for the frontend checks, @lucianmurmurache! Thanks so much for the work here. Really is heartening to have so much work going into a11y now 😊

**Pull request merged**